tests/integration/operatorhub/cmd_service_test.go

```diff
@@ -167,58 +228,34 @@ spec:
-
-			stdOut := helper.CmdShouldPass("odo", "service", "create", etcdOperator, "--crd", "EtcdCluster", "--dry-run")
-
-			// change the metadata.name from example to example3 so that we can run tests parallely
-			lines := strings.Split(stdOut, "\n")
-			for i, line := range lines {
-				if strings.Contains(line, "name: example") {
-					lines[i] = strings.Replace(lines[i], "example", "example3", 1)
-				}
-			}
-			stdOut = strings.Join(lines, "\n")
-
-			// stdOut contains the yaml specification. Store it to a file
-			randomFileName := helper.RandString(6) + ".yaml"
-			fileName := filepath.Join("/tmp", randomFileName)
-			if err := ioutil.WriteFile(fileName, []byte(stdOut), 0644); err != nil {
-				fmt.Printf("Could not write yaml spec to file %s because of the error %v", fileName, err.Error())
-			}
 
-			// now create operator backed service
-			helper.CmdShouldPass("odo", "service", "create", "--from-file", fileName)
+		JustBeforeEach(func() {
+			preSetup()
+		})
 
-			// now verify if the pods for the operator have started
-			pods := helper.CmdShouldPass("oc", "get", "pods", "-n", CI_OPERATOR_HUB_PROJECT)
-			// Look for pod with example name because that's the name etcd will give to the pods.
-			etcdPod := regexp.MustCompile(`example3-.[a-z0-9]*`).FindString(pods)
+		JustAfterEach(func() {
+			cleanPreSetup()
+		})
 
-			ocArgs := []string{"get", "pods", etcdPod, "-o", "template=\"{{.status.phase}}\"", "-n", CI_OPERATOR_HUB_PROJECT}
-			helper.WaitForCmdOut("oc", ocArgs, 1, true, func(output string) bool {
-				return strings.Contains(output, "Running")
-			})
+		It("should list the services if they exist", func() {
+			createEtcdClusterService()
 
-			stdOut = helper.CmdShouldPass("odo", "service", "list")
+			stdOut := helper.CmdShouldPass("odo", "service", "list")
 			Expect(stdOut).To(ContainSubstring("example"))
 			Expect(stdOut).To(ContainSubstring("EtcdCluster"))
 
 			// now check for json output
 			jsonOut := helper.CmdShouldPass("odo", "service", "list", "-o", "json")
-			helper.MatchAllInOutput(jsonOut, []string{"\"apiVersion\": \"etcd.database.coreos.com/v1beta2\"", "\"kind\": \"EtcdCluster\"", "\"name\": \"example3\""})
+			helper.MatchAllInOutput(jsonOut, []string{"\"apiVersion\": \"etcd.database.coreos.com/v1beta2\"", "\"kind\": \"EtcdCluster\"", "\"name\": \"example\""})
 
 			// Delete the pods created. This should idealy be done by `odo
 			// service delete` but that's not implemented for operator backed
 			// services yet.
-			helper.CmdShouldPass("oc", "delete", "EtcdCluster", "example3")
+			helper.CmdShouldPass("oc", "delete", "EtcdCluster", "example")
 
 			// Now let's check the output again to ensure expected behaviour
-			stdOut = helper.CmdShouldFail("odo", "service", "list")
-			jsonOut = helper.CmdShouldFail("odo", "service", "list", "-o", "json")
+			stdOut = helper.CmdShouldFail("odo", "service", "list", "--project", project)
```

Comment:
Project has been set in preSetup, So calling it here through `--project` makes no sense to me. Please check the similar occurence throughout the changes.

Reply:
https://github.com/openshift/odo/pull/3239#discussion_r429188003